rocky/reports/views/base.py

```diff
@@ -109,11 +111,16 @@ def get_report_types_for_aggregate_report(
             report_types[option] = self.get_report_types_for_generate_report(reports)
         return report_types
 
-    def get_required_optional_plugins(self, plugin_ids: Dict[str, Set[str]]) -> Dict[str, Plugin]:
+    def get_required_optional_plugins(self, plugin_ids: Dict[str, Set[str]]) -> (Dict[str, Plugin], Dict[str, bool]):
         plugins = {}
         for plugin, plugin_ids in plugin_ids.items():
             plugins[plugin] = [get_katalogus(self.organization.code).get_plugin(plugin_id) for plugin_id in plugin_ids]
-        return plugins
+
+        all_plugins_enabled = {
+            plugin_type: all(plugin.enabled for plugin in plugin_list) for plugin_type, plugin_list in plugins.items()
```

Comment:
I think you can also put the entire generator in `all`, potentially enhancing its laziness by causing it to stop at the first falsy value

Reply:
The original expression checks if all plugins across all plugin lists are enabled, while the suggestion expression checks if all plugins within each individual plugin list are enabled. So the current implementation is already correct 👍